addons/crm/models/crm_lead.py

```diff
@@ -234,30 +234,19 @@ def _onchange_partner_id(self):
         values = self._onchange_partner_id_values(self.partner_id.id if self.partner_id else False)
         self.update(values)
 
-    @api.model
-    def _onchange_user_values(self, user_id):
-        """ returns new values when user_id has changed """
-        if user_id and self._context.get('team_id'):
-            team = self.env['crm.team'].browse(self._context['team_id'])
-            if user_id in team.member_ids.ids:
-                return {}
-        team_id = self.env['crm.team']._get_default_team_id(user_id=user_id)
-        return {'team_id': team_id}
-
     @api.onchange('user_id')
     def _onchange_user_id(self):
         """ When changing the user, also set a team_id or restrict team id to the ones user_id is member of. """
-        values = self._onchange_user_values(self.user_id.id)
-        self.update(values)
+        # Keep team if commercial belongs to it
+        if self.user_id <= (self.team_id.member_ids | self.team_id.user_id):
+            return
+        self.team_id = self.env['crm.team']._get_default_team(self.user_id)
```

Comment:
if self.user_id > (self.team_id.member_ids | self.team_id.user_id):
         self.team_id = self.env['crm.team']._get_default_team(self.user_id)

don't need of `return`

Reply:
Your code would never execute. It should be:

```py
if not self.user_id <= (self.team_id.member_ids | self.team_id.user_id):
    self.team_id = self.env['crm.team']._get_default_team(self.user_id)
```

In any case, the result would be the same, so it's not a big deal IMHO.